nixos/modules/system/boot/luksroot.nix

```diff
@@ -118,6 +118,11 @@ let
     do_open_passphrase() {
         local passphrase
 
+        # Return if already unlocked
+        if [ -b "/dev/mapper/${name}" ]; then
+          return
+        fi
```

Comment:
Use `cryptsetup status ${name}` instead, it will exit with 0 if it exists (with 4 if it doesn't)